### PR TITLE
Use passfile in the primary_conninfo instead of password

### PR DESCRIPTION
**Pull request by CyberDem0n**
Fixed a few minor issues related to the #1134 and #1122
Close https://github.com/zalando/patroni/issues/1185

### CODE REVIEW

File: patroni/postgresql/__init__.py [hunk 1]
```diff
@@ -645,12 +631,10 @@ def controldata(self):
...
     def get_replication_connection_cursor(self, host='localhost', port=5432, database=None, **kwargs):
-        replication = self.config.replication
-        extra_kwargs = {k: v for k, v in replication.items()
-                        if k not in ('username', 'password', 'connect_timeout', 'options')}
-        with get_connection_cursor(host=host, port=int(port), database=database or self._database, replication=1,
-                                   user=replication['username'], password=replication.get('password'),
-                                   connect_timeout=3, options='-c statement_timeout=2000', **extra_kwargs) as cur:
+        conn_kwargs = self.config.replication.copy()
+        conn_kwargs.update(host=host, port=int(port), database=database or self._database, connect_timeout=3,
+                           user=conn_kwargs.pop('username', None), replication=1, options='-c statement_timeout=2000')
```
Comment (erthalion) [inline patroni/postgresql/__init__.py]: Looks like before lack of `username` key will cause an exception, which is not the case now. Is it intentional?

```diff
-user=replication['username']
+user=conn_kwargs.pop('username', None)
```

Comment (CyberDem0n) [reply patroni/postgresql/__init__.py]: Replication user name is mandatory, so the answer is no.
It should be just `conn_kwargs.pop('username')`.

File: patroni/postgresql/config.py [hunk 2]
```diff
@@ -300,28 +315,26 @@ def write_postgresql_conf(self, configuration=None):
         if 'custom_conf' not in self._config and not os.path.exists(self._postgresql_base_conf):
             os.rename(self._postgresql_conf, self._postgresql_base_conf)
 
-        with open(self._postgresql_conf, 'w') as f:
-            os.chmod(self._postgresql_conf, stat.S_IWRITE | stat.S_IREAD)
-            f.write(self._CONFIG_WARNING_HEADER)
-            f.write("include '{0}'\n\n".format(self._config.get('custom_conf') or self._postgresql_base_conf_name))
+        with ConfigWriter(self._postgresql_conf) as f:
+            include = self._config.get('custom_conf') or self._postgresql_base_conf_name
+            f.writeline("include '{0}'\n".format(ConfigWriter.escape(include)))
```
Comment (erthalion) [inline patroni/postgresql/config.py]: There is already one `\n` inside `writeline` method.

File: patroni/postgresql/config.py [hunk 3]
```diff
@@ -154,6 +154,37 @@ def mtime(filename):
...
+        self.writeline('# Do not edit this file manually!\n# It will be overwritten by Patroni!')
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self._fd:
+            self._fd.close()
+
+    def writeline(self, line):
+        self._fd.write(line)
+        self._fd.write('\n')
```
Comment (erthalion) [inline patroni/postgresql/config.py]: Why not one write with `{line}\n`?

**Comment by erthalion** [comment]
:+1:

File: patroni/postgresql/config.py (hunk 2 above)
Comment (CyberDem0n) [reply patroni/postgresql/config.py]: Originally there were two: https://github.com/zalando/patroni/pull/1194/files#diff-a534a7e0f2c46ba53349f820d9014468L306, therefore I am adding the second one.

File: patroni/postgresql/config.py (hunk 3 above)
Comment (CyberDem0n) [reply patroni/postgresql/config.py]: Just avoiding useless string concatenation.

**Comment by erthalion** [comment]
:+1:

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**